Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -770,6 +842,202 @@ def latest_version(
       nil
     end
     # rubocop:enable Metrics/CyclomaticComplexity
+
+    # Identifies the latest version of the resource in a given Formulae and returns a Hash containing
+    # the version information for a resource. Returns a nil value if a latest version
+    # couldn't be found for a given resource.
+    sig {
+      params(
+        resource: Resource,
+        json:     T::Boolean,
+        debug:    T::Boolean,
+        quiet:    T::Boolean,
+        verbose:  T::Boolean,
+      ).returns(Hash)
+    }
+    def resource_version(
+      resource,
+      json: false,
+      debug: false,
+      quiet: false,
+      verbose: false
+    )
+      has_livecheckable = resource.livecheckable?
+
+      if debug
+        puts "\n\n"
+        puts "Resource:         #{resource.name}"
+        puts "Livecheckable?:   #{has_livecheckable ? "Yes" : "No"}"
+      end
+
+      resource_version_info = {}
+
+      livecheck = resource.livecheck
+      livecheck_url = has_livecheckable ? livecheck.url : resource.url
```

Comment:
```suggestion
      livecheck_url = livecheck.url
```

As seen in `#latest_version`, the `livecheck_url` value should simply correspond to `livecheck.url` (i.e., we store the value in a variable to avoid repeatedly making a function call wherever `livecheck.url` is used). When it's set, it's the symbol/string from the `livecheck` block, otherwise it's the default `nil` value. That is to say, this shouldn't be falling back to the resource `url` because we handle that in subsequent code (i.e., when defining the `urls` variable).

Reply:
Yes, you are right. I had to add this check initially when `@livecheck` was `nil` in `Resource` DSL (before https://github.com/Homebrew/brew/pull/13753) to avoid getting error on resources which had no `livecheck` blocks.